src/models.py

```diff
@@ -1,19 +1,238 @@
 from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import Column, ForeignKey, Integer, String, Date
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+from sqlalchemy_utils import PasswordType
+import enum
 
 db = SQLAlchemy()
 
 class User(db.Model):
+    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    username = db.Column(db.String(40), nullable=False, unique=True)
+    password = db.Column(db.String(40), nullable=False, unique=False)
+    email = db.Column(db.String(320), unique=True, nullable=False)
+    name = db.Column(db.String(120), unique=False, nullable=False)
+    social = db.relationship('Social', backref='user', lazy=True)
+    is_active = db.Column(db.Boolean, default=True)
+    
+    def __repr__(self):
+        return f"User {self.username}"
+
+    def serialize(self):
+        return {
+            "id": self.id,
+            "username": self.username, 
+            "email": self.email,
+            "name": self.name
+        }
+    
+    def get_user(new_id):
+        user= User.query.filter_by(id=new_id, is_active=True).first()        
+        if user is None:
+            return None
+        else:       
+            return user.serialize()
+    
+    def update_user(new_id,body):
+     
+        user= User.query.filter_by(id=new_id, is_active=True).first() 
+        if user is None:
+            return None
+        else:
+            if "username" in body:
+                user.username = body["username"]
+            if "email" in body:
+                user.email = body["email"]
+            if "name" in body:
+                user.name = body["name"]
+            if "password" in body:
+                user.password = body["password"]
+            db.session.commit()
+            return "user with id: " + str(user.id) + " updated"
```

Comment:
```suggestion
        if "username" in body:
            user.username = body["username"]
        if "email" in body:
            user.email = body["email"]
        if "name" in body:
            user.name = body["name"]
        if "password" in body:
            user.password = body["password"]
        db.session.commit()
        return user
```
Después de un return no hace falta un else y no ponerlo mejora un poco la legibilidad

Es bueno evitar los mensajes de exito como strings, si una función ha hecho lo que debe hacer puede o devolver el objeto que ha generado o un True/False en función de si ha ido bien, los mensajes de éxito o error son difíciles de parsear y mantener.